soda_machine.rb

```diff
@@ -7,12 +9,38 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+
+    count = 0
+
+    @sodas.each { count += 1 }
+
+    return count
+
```

Comment:
Ruby has an inbuilt method called ```count``` which can be applied directly to arrays and strings. So you can call the count method directly on the @sodas arrray as shown below:
``` ruby
def current_inventory_count
    @sodas.count
end
```